components/ds3231/ds3231.c

```diff
@@ -87,6 +87,30 @@ static uint8_t dec2bcd(uint8_t val)
     return ((val / 10) << 4) + (val % 10);
 }
 
+#include <stdio.h>
+
+// Function to check if a year is a leap year
+int isLeapYear(int year) { return (year % 4 == 0 && year % 100 != 0) || (year % 400 == 0); }
+
+// Function to convert year, month, and day to days since January 1st
+int daysSinceJanuary1st(int year, int month, int day)
```

Comment:
1. No CamelCase allowed
2. It's static
3. It's inline

Reply:
This function is updated to static inline